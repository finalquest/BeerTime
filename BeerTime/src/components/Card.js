import React from 'react';
import {
  View,
  Image,
  StyleSheet,
} from 'react-native';

import { BeerProptype } from '../utils/PropTypes';
import Label from './Label';
import PropertyLabel from './PropertyLabel';


const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    height: 150,
    backgroundColor: 'white',
    margin: 20,
    padding: 10,
    borderRadius: 4,
    borderColor: 'gray',
    borderWidth: 1,
  },
  imageContainer: {
    flex: 0.2,
    marginLeft: 5,
  },
  image: {
    resizeMode: 'contain',
    flex: 1,
  },
  bodyContainer: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
    marginHorizontal: 5,
  },
  name: {
    fontSize: 25, textAlign: 'center',
  },
});


const Card = ({ beer }) => {
  const {
    image_url: imageUrl, name, tagline, first_brewed: firstBrewed,
  } = beer;
  return (
    <View style={styles.container}>
      <View style={styles.imageContainer}>
        <Image
          style={styles.image}
          source={{ uri: imageUrl }}
        />
      </View>
      <View style={styles.bodyContainer}>
        <Label style={styles.name} ellipsizeMode="tail" numberOfLines={2}>{name}</Label>
        <Label style={{ fontSize: 11 }}>{tagline}</Label>
        <PropertyLabel
          name="First brew date"
          style={{ marginVertical: 10 }}
          value={firstBrewed}
          separator=":"
        />
      </View>
    </View>
  );
};
Card.propTypes = {
  beer: BeerProptype.isRequired,
};
Card.defaultProps = {
};

export default Card;
// export default ProductBox;
