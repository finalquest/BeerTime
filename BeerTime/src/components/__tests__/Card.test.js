import React from 'react';
import { shallow } from 'enzyme';

import Card from '../Card';

describe('Card test suite', () => {
  it('renders as expected', () => {
    const wrapper = shallow(<Card beer={{ image_url: 'testUrl' }} />);
    expect(wrapper).toMatchSnapshot();
  });
});
