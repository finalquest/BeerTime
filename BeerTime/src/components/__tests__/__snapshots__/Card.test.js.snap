// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card test suite renders as expected 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "borderColor": "gray",
      "borderRadius": 4,
      "borderWidth": 1,
      "flexDirection": "row",
      "height": 150,
      "margin": 20,
      "padding": 10,
    }
  }
>
  <View
    style={
      Object {
        "flex": 0.2,
        "marginLeft": 5,
      }
    }
  >
    <Image
      source={
        Object {
          "uri": "testUrl",
        }
      }
      style={
        Object {
          "flex": 1,
          "resizeMode": "contain",
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "marginHorizontal": 5,
      }
    }
  >
    <Label
      ellipsizeMode="tail"
      numberOfLines={2}
      style={
        Object {
          "fontSize": 25,
          "textAlign": "center",
        }
      }
      uppercase={false}
    />
    <Label
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 11,
        }
      }
      uppercase={false}
    />
    <PropertyLabel
      Component={[Function]}
      ellipsizeMode="tail"
      fontSize={14}
      name="First brew date"
      nameStyles={
        Object {
          "fontWeight": "400",
          "marginRight": 3,
          "textAlign": "center",
        }
      }
      numberOfLines={1}
      separator=":"
      style={
        Object {
          "marginVertical": 10,
        }
      }
      valueStyles={
        Object {
          "flexShrink": 1,
          "textAlign": "center",
        }
      }
    />
  </View>
</View>
`;
